x-pack/legacy/plugins/ml/server/models/calendar/calendar_manager.ts

```diff
@@ -6,17 +6,46 @@
 
 import _ from 'lodash';
 import Boom from 'boom';
-import { EventManager } from './event_manager';
+import { EventManager, CalendarEvent } from './event_manager';
+
+interface BasicCalendar {
+  job_ids: string[];
+  description?: string;
+  events: CalendarEvent[];
+}
+
+export interface Calendar extends BasicCalendar {
+  calendar_id: string;
+}
+
+export interface FormCalendar extends BasicCalendar {
+  calendarId: string;
+}
 
 export class CalendarManager {
-  constructor(callWithRequest) {
-    this.callWithRequest = callWithRequest;
-    this.eventManager = new EventManager(callWithRequest);
+  private isLegacy: boolean;
+  private client: any;
+  private eventManager: any;
+
+  constructor(isLegacy: boolean, client: any) {
+    this.isLegacy = isLegacy;
+    this.client = client;
+    this.eventManager = new EventManager(isLegacy, client);
   }
 
-  async getCalendar(calendarId) {
+  async getCalendar(calendarId: string) {
     try {
-      const resp = await this.callWithRequest('ml.calendars', { calendarId });
+      let resp;
+      if (this.isLegacy === true) {
```

Comment:
as per the previous comment, having these checks inside the class seems unnecessary and means we'll have to change them again at some point in the future.

Reply:
I've just seen the comment on this PR referring to the need for compatibility with the still legacy job service, so please ignore the suggestion of moving these checks out side of the class.
I still think the check could be moved to the constructor, so it's only happening once also `isLegacy` wouldn't need to be stored.